packages/vscode-ext/client/tsconfig.json

```diff
@@ -1,12 +0,0 @@
-{
-  "compilerOptions": {
-    "module": "commonjs",
-    "target": "es2020",
-    "lib": ["ES2020", "dom"],
-    "outDir": "out",
-    "rootDir": "src",
-    "sourceMap": true
-  },
-  "include": ["src"],
-  "exclude": ["node_modules", ".vscode-test"]
-}
```

Comment:
Are you sure this should be removed? Top-level tsconfig.json still mentions it in references, and typescript is still needed for type checks.

Reply:
My assumption was that top-level tsconfig specifies extension host environment, and server tsconfig does so for node~ worker environment.

I kind of didn't understand the need for 3 tsconfigs in one project, and started deleting tautologies)
Should I get it back or probably remove reference in the toplevel instead?

Reply:
I originally copied the 3 tsconfigs structure from here: https://github.com/microsoft/vscode-extension-samples/tree/main/lsp-sample (which is linked from the official [Language Server Extension Guide](https://code.visualstudio.com/api/language-extensions/language-server-extension-guide)).

Looks like their tsconfigs are a different for client and for server, and they have separate package.json files too.

Feel free to clean that all up. My guess is that `{client,server}/src` could be merged in a single `src/{client,server}` with a single top-level tsconfig.json.

Reply:
Well, did pretty much that. Basically, since I use cjs for both outputs (and it works!) I left only one config for everything, and merged the trees.